ibis/pyspark/timecontext.py

```diff
@@ -26,10 +26,40 @@ def filter_by_time_context(
 
     begin, end = timecontext
     if TIME_COL in df.columns:
-        return df.filter((F.col(TIME_COL) >= begin) & (F.col(TIME_COL) < end))
+        # for py3.8, underlying spark type converter calls utctimetuple()
```

Comment:
I am not sure this is correct, I am able to filter by pd.Timestamp:

```
(ibis-ci-testing) Lis-MacBook-Pro:ibis icexelloss$ pyspark
Python 2.7.15 (default, Jun 17 2018, 12:46:58) 
[GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
20/10/02 14:30:11 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
Using Spark's default log4j profile: org/apache/spark/log4j-defaults.properties
Setting default log level to "WARN".
To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
/usr/local/lib/python2.7/site-packages/pyspark/context.py:225: DeprecationWarning: Support for Python 2 and Python 3 prior to version 3.6 is deprecated as of Spark 3.0. See also the plan for dropping Python 2 support at https://spark.apache.org/news/plan-for-dropping-python-2-support.html.
  DeprecationWarning)
Welcome to
      ____              __
     / __/__  ___ _____/ /__
    _\ \/ _ \/ _ `/ __/  '_/
   /__ / .__/\_,_/_/ /_/\_\   version 3.0.1
      /_/

Using Python version 2.7.15 (default, Jun 17 2018 12:46:58)
SparkSession available as 'spark'.
>>> df = spark.range(10)
df = d>>> df = df.withColumn('time', df['id'].cast('timestamp'))
>>> df
DataFrame[id: bigint, time: timestamp]
>>> df.toPandas()
   id                time                                                       
0   0 1969-12-31 19:00:00
1   1 1969-12-31 19:00:01
2   2 1969-12-31 19:00:02
3   3 1969-12-31 19:00:03
4   4 1969-12-31 19:00:04
5   5 1969-12-31 19:00:05
6   6 1969-12-31 19:00:06
7   7 1969-12-31 19:00:07
8   8 1969-12-31 19:00:08
9   9 1969-12-31 19:00:09
>>> import pandas as pd
>>> df.filter(df['time'] > pd.Timestamp("20200101"))
DataFrame[id: bigint, time: timestamp]
>>> df.filter(df['time'] > pd.Timestamp("20200101")).show()
+---+----+
| id|time|
+---+----+
+---+----+
```

Reply:
You need to set up timezone to trigger this bug, see https://github.com/pandas-dev/pandas/issues/32174.
Try `pd.Timestamp("20200101", tz="

Reply:
This is a fix to address the failing py3.8 tests. Tests will fail for py3.8 with `TypeError: an integer is required`

Reply:
I still prefer to use `pd.Timestamp` instead of pydatetime. Perhaps drop the timezone here?

Please also refer to the pandas issue in the comment.

Reply:
I added the pandas issue inline as comment. But dropping timezone may cause error because spark may interpret the timestamp without a timezone with session Timezone (or os.environ['TZ']) which may cause unexpected behavior.  So I guess we would better keep tz, if it is specified?  This gurantees timezone will not be misinterpreted in execution

Reply:
Hmm...Does Pyspark interpret py_datetime as UTC timestamp not session local and intepret pd.Timestamp without timezone as session local?